src/webgpu/api/validation/render_pass/loadOp_and_storeOp.spec.ts

```diff
@@ -0,0 +1,145 @@
+}
+
+g.test('load_op_store_op_and_read_only')
+  .params(u =>
+    u
+      .beginSubcases() // Note: It's easier to debug if you comment this line out as you can then run an individual case.
+      .combine('aspect', [undefined, 'all', 'depth-only', 'stencil-only'] as GPUTextureAspect[])
+      .combine('depthReadOnly', [undefined, true, false])
+      .combine('depthLoadOp', [undefined, 'clear', 'load'] as GPULoadOp[])
+      .combine('depthStoreOp', [undefined, 'discard', 'store'] as GPUStoreOp[])
+      .combine('stencilReadOnly', [undefined, true, false])
+      .combine('stencilLoadOp', [undefined, 'clear', 'load'] as GPULoadOp[])
+      .combine('stencilStoreOp', [undefined, 'discard', 'store'] as GPUStoreOp[])
+  )
+  .fn(t => {
+    const {
+      aspect,
+      depthReadOnly,
+      depthLoadOp,
+      depthStoreOp,
+      stencilReadOnly,
+      stencilLoadOp,
+      stencilStoreOp,
+    } = t.params;
+
+    const format = getTextureFormatForAspect(aspect);
+    const depthAttachment = t.trackForCleanup(
+      t.device.createTexture({
+        format,
+        size: { width: 1, height: 1, depthOrArrayLayers: 1 },
+        usage: GPUTextureUsage.RENDER_ATTACHMENT,
+      })
+    );
+    const depthAttachmentView = depthAttachment.createView({
+      ...(!!aspect && { aspect }),
+    });
+
+    const encoder = t.device.createCommandEncoder();
+    const renderPassDescriptor: GPURenderPassDescriptor = {
+      colorAttachments: [],
+      depthStencilAttachment: {
+        view: depthAttachmentView,
+        depthLoadOp,
+        depthStoreOp,
+        depthReadOnly,
+        stencilLoadOp,
+        stencilStoreOp,
+        stencilReadOnly,
+      },
+    };
+    const pass = encoder.beginRenderPass(renderPassDescriptor);
+    pass.end();
+
+    const hasDepthSettings = !!depthLoadOp && !!depthStoreOp && !depthReadOnly;
+    const hasStencilSettings = !!stencilLoadOp && !!stencilStoreOp && !stencilReadOnly;
+    const hasDepth = !aspect || aspect === 'all' || aspect === 'depth-only';
+    const hasStencil = !aspect || aspect === 'all' || aspect === 'stencil-only';
+
+    const goodAspectCombo =
+      (!aspect || aspect === 'all' ? !depthReadOnly === !stencilReadOnly : true) &&
+      (hasDepthSettings ? hasDepth : true) &&
+      (hasStencilSettings ? hasStencil : true);
+
+    const hasBothDepthOps = !!depthLoadOp && !!depthStoreOp;
+    const hasBothStencilOps = !!stencilLoadOp && !!stencilStoreOp;
+    const hasNeitherDepthOps = !depthLoadOp && !depthStoreOp;
+    const hasNeitherStencilOps = !stencilLoadOp && !stencilStoreOp;
+    const goodDepthCombo = hasDepth && !depthReadOnly ? hasBothDepthOps : hasNeitherDepthOps;
+    const goodStencilCombo =
+      hasStencil && !stencilReadOnly ? hasBothStencilOps : hasNeitherStencilOps;
+
+    const shouldError = !goodAspectCombo || !goodDepthCombo || !goodStencilCombo;
+
+    t.expectValidationError(() => {
+      encoder.finish();
+    }, shouldError);
+  });
+
+g.test('depthClearValue')
```

Comment:
So couple thoughts here, initially I saw this and thought that the test name doesn't really make sense with the name of the suite so I was looking around to see whether there was another more suitable suite for it. I then came across https://github.com/gpuweb/cts/blob/main/src/webgpu/api/validation/render_pass/render_pass_descriptor.spec.ts#L885 which is a very similar test, and also https://github.com/gpuweb/cts/blob/main/src/webgpu/api/validation/render_pass/render_pass_descriptor.spec.ts#L811. So I was thinking that (1) we should consolidate those tests so that we don't have duplicates, and that (2) maybe we can have a `depth_stencil_attachment` suite under `render_pass` and keep all these tests in there?

Sorry I didn't see those tests earlier! D:

Reply:
agreed. done